js/foundation.offcanvas.js

```diff
@@ -132,7 +132,7 @@ class OffCanvas {
     } else {
       this.isRevealed = false;
       this.$element.attr('aria-hidden', 'true');
-      this.$element.on({
+      this.$element.off('open.zf.trigger toggle.zf.trigger').on({
```

Comment:
Because currently, if you trigger this branch in the else twice (for example by resizing across 2 sizes that both are not revealed) the trigger gets double applied, and counteracts itself, resulting in the behavior from #9695 where the 2 event handlers cancel themselves out